src/pebble.py

```diff
@@ -125,16 +125,16 @@ def check_irc_bridge_ready() -> ops.pebble.CheckDict:
     return check.to_dict()
 
 
-def replan_nginx(container: ops.model.Container, main_unit_address: str) -> None:
-    """Replan Synapse NGINX service and regenerate configuration.
+def restart_nginx(container: ops.model.Container, main_unit_address: str) -> None:
+    """Restart Synapse NGINX service and regenerate configuration.
 
     Args:
         container: Charm container.
         main_unit_address: Main unit address to be used in configuration.
     """
     container.add_layer("synapse-nginx", _nginx_pebble_layer(), combine=True)
     synapse.generate_nginx_config(container=container, main_unit_address=main_unit_address)
-    container.replan()
+    container.restart(synapse.SYNAPSE_NGINX_SERVICE_NAME)
```

Comment:
Won't this trigger more restarts and cause more downtime that a replan?

Reply:
Using the replan doesn't get changes in the configuration file and when scaling, the file changes. I noticed that while deploying Synapse with two units and one of the units didn't have the right configuration.
It would be nice to have something like container.reload or we can work in a further PR to reload NGINX using exec.
WDYT?

Reply:
Oh, makes sense. I think that if this is good enough I wouldn't complicate it further. But those are definitely good options